### PR TITLE
Datatrails: Sync layout with URL values

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

This PR syncs the selected layout (single, rows or grid) with the URL so it's kept when switching tabs, selecting a new metric or refreshing the page. It's also stored when creating a bookmark. 

**Why do we need this feature?**

Improves UX when switching action tabs. 

**Who is this feature for?**

Users of data trails / explore metrics. 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trails/ActionTabs/LayoutSwitcher.tsx [hunk 1]
```diff
@@ -14,23 +16,25 @@ export type LayoutType = 'single' | 'grid' | 'rows';
...
       <Field label="View">
-        <RadioButtonGroup options={options} value={active} onChange={model.onLayoutChange} />
+        <RadioButtonGroup options={options} value={layout} onChange={model.onLayoutChange} />
       </Field>
     );
   }
 
   public onLayoutChange = (active: LayoutType) => {
-    this.setState({ active });
+    locationService.partial({ layout: active });
```
Comment (torkelo) [inline public/app/features/trails/ActionTabs/LayoutSwitcher.tsx:30]: Can we let url sync system do this ? We need to support data trails in the embedded (drawer) mode where url sync is disabled

File: public/app/features/trails/MetricScene.tsx [hunk 2]
```diff
@@ -40,21 +41,23 @@ export interface MetricSceneState extends SceneObjectState {
   body: MetricGraphScene;
   metric: string;
   actionView?: string;
+  layout: LayoutType;
```
Comment (torkelo) [inline public/app/features/trails/MetricScene.tsx:44]: Any specific reason why you added this to the MetricScene and not just added URL sync to LayoutSwitcher?

**Review by torkelo — APPROVED** [review]
looks ok, just wonder why the state/urlsync was moved to the MetricScene

File: public/app/features/trails/ActionTabs/LayoutSwitcher.tsx [hunk 3]
```diff
@@ -1,36 +1,43 @@
...
       <Field label="View">
-        <RadioButtonGroup options={options} value={active} onChange={model.onLayoutChange} />
+        <RadioButtonGroup options={options} value={layout} onChange={model.onLayoutChange} />
       </Field>
     );
   }
 
   public onLayoutChange = (active: LayoutType) => {
-    this.setState({ active });
+    this.getMetricScene().setState({ layout: active });
```
Comment (torkelo) [inline public/app/features/trails/ActionTabs/LayoutSwitcher.tsx:33]: would be cool to store in local storage so we can remember this as a default preference

**Review by darrenjaneczek — APPROVED** [review]

**Pull request merged**